src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -359,6 +352,42 @@ internal static bool TryCreate(SyntheticBoundNodeFactory F, MethodSymbol method,
+
+            void validateIgnoredBuilderType(SyntheticBoundNodeFactory F, NamedTypeSymbol returnType, object builderArgument, bool isGeneric)
+            {
+                var ignoredBuilderType = ValidateBuilderType(F, builderArgument, returnType.DeclaredAccessibility, isGeneric);
+
+                if ((object)ignoredBuilderType != null)
+                {
+                    if (isGeneric)
+                    {
+                        var resultType = returnType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics.Single().Type;
+                        ignoredBuilderType = ignoredBuilderType.ConstructedFrom.Construct(resultType);
+                    }
+
+                    GetCustomCreateMethod(F, ignoredBuilderType, forOverride: false);
+                    GetCustomTaskProperty(F, ignoredBuilderType, returnType);
+
+                    _ = TryGetBuilderMember(F,
+                            isGeneric ? WellKnownMember.System_Runtime_CompilerServices_AsyncTaskMethodBuilder_T__SetException : WellKnownMember.System_Runtime_CompilerServices_AsyncTaskMethodBuilder__SetException,
+                            ignoredBuilderType, customBuilder: true, out MethodSymbol _) &&
+                        TryGetBuilderMember(F,
+                            isGeneric ? WellKnownMember.System_Runtime_CompilerServices_AsyncTaskMethodBuilder_T__SetResult : WellKnownMember.System_Runtime_CompilerServices_AsyncTaskMethodBuilder__SetResult,
+                            ignoredBuilderType, customBuilder: true, out MethodSymbol _) &&
+                        TryGetBuilderMember(F,
+                            isGeneric ? WellKnownMember.System_Runtime_CompilerServices_AsyncTaskMethodBuilder_T__AwaitOnCompleted : WellKnownMember.System_Runtime_CompilerServices_AsyncTaskMethodBuilder__AwaitOnCompleted,
+                            ignoredBuilderType, customBuilder: true, out MethodSymbol _) &&
+                        TryGetBuilderMember(F,
+                           isGeneric ? WellKnownMember.System_Runtime_CompilerServices_AsyncTaskMethodBuilder_T__AwaitUnsafeOnCompleted : WellKnownMember.System_Runtime_CompilerServices_AsyncTaskMethodBuilder__AwaitUnsafeOnCompleted,
+                           ignoredBuilderType, customBuilder: true, out MethodSymbol _) &&
+                        TryGetBuilderMember(F,
+                           isGeneric ? WellKnownMember.System_Runtime_CompilerServices_AsyncTaskMethodBuilder_T__Start_T : WellKnownMember.System_Runtime_CompilerServices_AsyncTaskMethodBuilder__Start_T,
+                           ignoredBuilderType, customBuilder: true, out MethodSymbol _) &&
+                        TryGetBuilderMember(F,
+                            isGeneric ? WellKnownMember.System_Runtime_CompilerServices_AsyncTaskMethodBuilder_T__SetStateMachine : WellKnownMember.System_Runtime_CompilerServices_AsyncTaskMethodBuilder__SetStateMachine,
+                            ignoredBuilderType, customBuilder: true, out MethodSymbol _);
```

Comment:
Could we use the existing `TryCreate()` method instead?

Reply:
It doesn't make much difference. I did initially ;-) I'll change it back.